labview/labview_rabbitmq.c

```diff
@@ -5,251 +5,170 @@
 	amqp_connection_state_t conn = amqp_new_connection();
 	// cast to int64_t so LabVIEW will receive it as 
 	// a 64-bit integer transfered by value
-	int64_t conn_intptr = (int64_t)conn;
-	return conn_intptr; 
+	int64_t conn_intptr = (int64_t) conn;
+	return conn_intptr;
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_close_connection(int64_t conn_intptr, LStrHandle error_description) {
+int lv_amqp_close_connection(int64_t conn_intptr, LStrHandle errorDescription)
+{
 	// cast back to amqp_connection_state_t opaque struct pointer
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
-	return lv_report_amqp_error(amqp_connection_close(conn, AMQP_REPLY_SUCCESS),"Connection close", error_description);
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
+	return lv_report_amqp_error(amqp_connection_close(conn, AMQP_REPLY_SUCCESS), "Connection close", errorDescription);
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-void lv_amqp_destroy_connection(int64_t conn_intptr) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+void lv_amqp_destroy_connection(int64_t conn_intptr)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
 	amqp_destroy_connection(conn);
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_channel_open(int64_t conn_intptr, uint16_t  channel, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+int lv_amqp_channel_open(int64_t conn_intptr, uint16_t channel, LStrHandle errorDescription)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
 	amqp_channel_open(conn, channel);
-	return lv_report_amqp_error(amqp_get_rpc_reply(conn), "Opening channel", error_description);
+	return lv_report_amqp_error(amqp_get_rpc_reply(conn), "Opening channel", errorDescription);
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_channel_close(int64_t conn_intptr, uint16_t  channel, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
-	return lv_report_amqp_error(amqp_channel_close(conn, channel, AMQP_REPLY_SUCCESS), "Closing channel", error_description);
+int lv_amqp_channel_close(int64_t conn_intptr, uint16_t channel, LStrHandle errorDescription)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
+	return lv_report_amqp_error(amqp_channel_close(conn, channel, AMQP_REPLY_SUCCESS), "Closing channel", errorDescription);
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_exchange_declare(int64_t conn_intptr, uint16_t  channel, char *exchange, char *exchangetype, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+int lv_amqp_exchange_declare(int64_t conn_intptr, uint16_t channel, char *exchange, char *exchangetype, uint8_t passive, LStrHandle errorDescription)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
 
-	amqp_boolean_t PASSIVE = 0;
+	amqp_boolean_t PASSIVE = passive;
```

Comment:
I saw on lv side you have not expose passive as a control when declaring new exchange, is it normaml?

Reply:
well, it's used in the newest version of the lv code. If not - the we will have a crash each time that the function is called.

Reply:
Yes I saw it is wired as a constant inside the create queue vi, but when you place the create queue vi in your code you can select this option without changing the vi of the palette